extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/devmode/ArcEndpointRecorder.java

```diff
@@ -0,0 +1,131 @@
+                    InjectableBean<?> injectableBean = it.next();
+                    if (kind != null && !kind.equals(injectableBean.getKind())) {
+                        it.remove();
+                    }
+                    if (scopeEndsWith != null && !injectableBean.getScope().getName().endsWith(scopeEndsWith)) {
+                        it.remove();
+                    }
+                    if (beanClassStartsWith != null
+                            && !injectableBean.getBeanClass().getName().startsWith(beanClassStartsWith)) {
+                        it.remove();
+                    }
+                }
+
+                JsonArrayBuilder array = Json.array();
+                for (InjectableBean<?> injectableBean : beans) {
+                    JsonObjectBuilder bean = Json.object();
+                    bean.put("id", injectableBean.getIdentifier());
+                    bean.put("kind", injectableBean.getKind().toString());
+                    bean.put("generatedClass", injectableBean.getClass().getName());
+                    bean.put("beanClass", injectableBean.getBeanClass().getName());
+                    JsonArrayBuilder types = Json.array();
+                    for (Type beanType : injectableBean.getTypes()) {
+                        types.add(beanType.getTypeName());
+                    }
+                    bean.put("types", types);
+                    JsonArrayBuilder qualifiers = Json.array();
+                    for (Annotation qualifier : injectableBean.getQualifiers()) {
+                        if (qualifier.annotationType().equals(Any.class) || qualifier.annotationType().equals(Default.class)) {
+                            qualifiers.add("@" + qualifier.annotationType().getSimpleName());
+                        } else {
+                            qualifiers.add(qualifier.toString());
+                        }
+                    }
+                    bean.put("qualifiers", qualifiers);
+                    bean.put("scope", injectableBean.getScope().getName());
+
+                    if (injectableBean.getDeclaringBean() != null) {
+                        bean.put("declaringBean", injectableBean.getDeclaringBean().getIdentifier());
+                    }
+                    if (injectableBean.getName() != null) {
+                        bean.put("name", injectableBean.getName());
+                    }
+                    if (injectableBean.isAlternative()) {
+                        bean.put("alternativePriority", injectableBean.getAlternativePriority());
+                    }
+                    if (injectableBean.isDefaultBean()) {
+                        bean.put("isDefault", true);
+                    }
+                    array.add(bean);
+                }
+                ctx.response().end(array.build());
+            }
+        };
+    }
+
+    public Handler<RoutingContext> createObserversHandler() {
+        return new Handler<RoutingContext>() {
```

Comment:
We should document this capability and the options you have in the CDI guide

Reply:
Yeah, good idea.